smart-contracts/advanced/multicall.md

```diff
@@ -0,0 +1,141 @@
+---
+description: >-
+  Multicall allows you to aggregate multiple contract reads into a single JSON-RPC request,
+  and execute multiple state-changing calls in a single transaction on the FVM.
+---
+
+# Multicall
+
+Multicall is a powerful tool that offers several advantages for interacting with smart contracts on the Filecoin Virtual Machine (FVM). 
+
+## Contract address
+
+Multicall has the same, precomputed address for all of the networks it is deployed on. 
+| Name             | Address                                      | Mainnet | Calibration |
+| ---------------- | -------------------------------------------- | ------- | ----------- |
+| [Multicall - Mainnet](https://filfox.info/en/address/0xcA11bde05977b3631167028862bE2a173976CA11?t=3) | `0xcA11bde05977b3631167028862bE2a173976CA11` | ✔️      | ❌          |
+| [Multicall - Calibration](https://calibration.filscan.io/en/address/0xcA11bde05977b3631167028862bE2a173976CA11/) | `0xcA11bde05977b3631167028862bE2a173976CA11` | ❌      | ✔️          |
+
+## Contract Deployments and ABI
+
+Multicall3 is deployed on over 100 chains at `0xcA11bde05977b3631167028862bE2a173976CA11`[^2].
+A sortable, searchable list of all chains it's deployed on can be found at https://multicall3.com/deployments.
+
+The ABI can be found on https://multicall3.com/abi, where it can be downloaded or copied to the clipboard in various formats, including:
+
+- Solidity interface.
+- JSON ABI, prettified.
+- JSON ABI, minified.
+- [ethers.js](https://docs.ethers.org/v5/) human readable ABI.
+- [viem](https://viem.sh/) human readable ABI.
+
+Alternatively, you can:
+
+- Download the ABI from the [releases](https://github.com/mds1/multicall/releases) page.
+- Copy the ABI from [Etherscan](https://etherscan.io/address/0xcA11bde05977b3631167028862bE2a173976CA11#code).
+- Install [Foundry](https://github.com/gakonst/foundry/) and run `cast interface 0xcA11bde05977b3631167028862bE2a173976CA11`.
+
+## Usage
```

Comment:
If I understand it correctly, builders need to call multicalls contract if they want to use the batch read and write to their own smart contract, right? So there will be no changes or extra work for builders to change their smart contract to be able to work with multicalls? 
If I am not wrong, maybe we can add some clarification and instruction here. 

**For example (Just an example with random write up, please change it accordingly )**

----------
To use Multicall3 to send batch contract read/write to your smart contract, you will need to:
1.  NO extra changes or adjustment of your own smart contract.
2. Import Multicall3 ABI in your code by doing ......
3. Caling Multicall3 with target contract, methods and params for multiple read/write
4. receiving results for multicalls and process in your code.  
-------------------

Reply:
Pushing back with some detail: the current usage section links to Multicall3 examples in multiple languages [Multicall3 examples folder](https://github.com/mds1/multicall/blob/main/examples) and has example usage code (for which function to interact in solidity with and what to do in ethers.js)

```
When directly interacting with the Multicall3 contract to batch calls, you'll typically use the `aggregate3` method. This method allows you to execute multiple contract calls in a single transaction. Here's an explanation of how it works, along with examples:
```

There does involve some changes in smart contract or whichever setup people were previously using, detailed by the above. What do you think? How do we make this more clear?

Reply:
Example: we can do:

- Point one
- Point two
- Point three

And then go into the details